Core/source/mobile-interface/endpoints.ts

```diff
@@ -64,8 +64,13 @@ export class Endpoints {
         const encryptedAtt = await PgpMsg.encrypt({ pubkeys: req.pubKeys, data: Buf.fromBase64Str(att.base64), filename: att.name, armor: false }) as OpenPGP.EncryptBinaryResult;
         encryptedAtts.push(new Att({ name: att.name, type: 'application/pgp-encrypted', data: encryptedAtt.message.packets.write() }))
       }
-      const encrypted = await PgpMsg.encrypt({ pubkeys: req.pubKeys, data: Buf.fromUtfStr(req.text), armor: true }) as OpenPGP.EncryptArmorResult;
-      return fmtRes({}, Buf.fromUtfStr(await Mime.encode({ 'text/plain': encrypted.data }, mimeHeaders, encryptedAtts)));
+      const signingPrv = await readArmoredKeyOrThrow(req.signingPrv.private);
```

Comment:
Looks ok except the `signingPrv` should be optional. If I supply it, message gets signed. If I don't, it won't.

Reply:
This is true only for the TypeScript side or also for the Swift side?